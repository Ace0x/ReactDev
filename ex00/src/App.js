import { useState, useRef } from "react";

var emls = [];
var pwords = [];

function App() {
  const nameRef = useRef(null);
  const surRef = useRef(null);
  const emRef = useRef(null);
  const pwRef = useRef(null);
  const emRef2 = useRef(null);
  const pwRef2 = useRef(null);
  const [name, setName] = useState("");
  const [surname, setSur] = useState("");
  const [email, setEm] = useState("");
  const [pword, setPw] = useState("");
  const [email2, setEm2] = useState("");
  const [pword2, setPw2] = useState("");

  function checkArr(_email, _pword) {
    var flag = false;
    for (var i = 0; i < emls.length; i++) {
      console.log(emls[i]);
      if (emls[i] === _email) {
        flag = true;
        break;
      }
    }
    if (!flag) {
      emls.push(_email);
      pwords.push(_pword);
    }
  }

  function log() {
    setEm2(emRef2.current.value);
    setPw2(pwRef2.current.value);
    if (login(email2, pword2)) {
      console.log("success!");
    }
  }

  function login(_email, _pword) {
    for (var i = 0; i < emls.length; i++) {
      if (emls[i] === _email) {
        if (pwords[i] === _pword) {
          return true;
        }
      }
    }
    return false;
  }

  function handleClick() {
    setName(nameRef.current.value);
    setSur(surRef.current.value);
    setEm(emRef.current.value);
    setPw(pwRef.current.value);
    checkArr(email, pword);
  }

  return (
    <div className="App">
      <h1>Register</h1>
      <h2>Name:</h2>
      <input ref={nameRef} type="text" id="name" name="name" />
      <h2>Surname :</h2>
      <input ref={surRef} type="text" id="surname" name="surname" />
      <h2>Email :</h2>
      <input ref={emRef} type="text" id="email" name="email" />
      <h2>Password :</h2>
      <input ref={pwRef} type="password" id="pword" name="pword" />
      <br></br>
      <button onClick={handleClick}> Submit</button>
      <h2>Contact:</h2>
      <p>
        {" "}
        {name} {surname}
      </p>
      <h2>Email:</h2>
      <p>{email}</p>

      <h1>Login</h1>
      <h2>Email :</h2>
      <input ref={emRef2} type="text" id="email2" name="email2" />
      <h2>Password :</h2>
      <input ref={pwRef2} type="password" id="pword2" name="pword2" />
      <button onClick={log}> Submit</button>
    </div>
  );
}

export default App;
