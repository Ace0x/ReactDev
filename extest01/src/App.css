.encompass {
    display: grid;
    grid-template-columns: 2;
    grid-template-rows: 1;
}

.first {
    border: 2px red solid;
    grid-column: 1;
    grid-row: 1;
    display: flex;
    justify-content: center;
    margin: 5px;
}

.second {
    border: 2px blue solid;
    grid-column: 2;
    grid-row: 1;
    display: flex;
    justify-content: center;
    margin: 5px;
}

Table {
    width: max-content;
}