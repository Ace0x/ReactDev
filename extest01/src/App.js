
import { useState } from "react";
import Papa from "papaparse";
import Table from "./list";
import evaluate_entry from "./MxSystem";

function App() {
  // State to store parsed data
  const [parsedData, setParsedData] = useState([]);

  //State to store table Column name
  const [tableRows, setTableRows] = useState([]);

  //State to store the values
  const [values, setValues] = useState([]);
  
  //State to store changed values
  const [mxValues, setMx] = useState([]);

  const changeHandler = (event) => {
    // Passing file data (event.target.files[0]) to parse using Papa.parse
    Papa.parse(event.target.files[0], {
      header: true,
      skipEmptyLines: true,
      complete: function (results) {
        const rowsArray = [];
        const valuesArray = [];

        // Iterating data to get column name and their values
        results.data.map((d) => {
          rowsArray.push(Object.keys(d));
          valuesArray.push(Object.values(d));
        });

        // Parsed Data Response in array format
        setParsedData(results.data);

        // Filtered Column Names
        setTableRows(rowsArray[0]);
        console.log("SETTING: " + tableRows);

        // Filtered Values
        setValues(valuesArray);

        let mxArray = [];
        console.log(valuesArray.length);
        console.log("Values:  " + valuesArray);
        for(let i = 0; i < valuesArray.length; i++)
        {
          console.log(valuesArray[i]);
          mxArray.push(evaluate_entry(valuesArray[i]));
        }
        setMx(mxArray);
      },
    });
    //Convert Values
 
  };

  return (
    <div>
      {/* File Uploader */}
      <input
        type="file"
        name="file"
        onChange={changeHandler}
        accept=".csv"
        style={{ display: "block", margin: "10px auto" }}
      />
      <br />
      <br />
      {/* Table */}
      <Table values = {values} tableRows={tableRows}/>
      <br />
      <Table values = {mxValues} tableRows={tableRows}/>
    </div>
  );
}

export default App;
