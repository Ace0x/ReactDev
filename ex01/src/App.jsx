import "./App.css";
import { useState } from "react";
import Papa from "papaparse";
import Table from "./table";

function App() {
  // State to store parsed data
  const [parsedData, setParsedData] = useState([]);
  //State to store table Column name
  const [tableRows, setTableRows] = useState([]);

  //State to store the values
  const [values, setValues] = useState([]);

  const changeHandler = (event) => {
    // Passing file data (event.target.files[0]) to parse using Papa.parse
    Papa.parse(event.target.files[0], {
      header: true,
      skipEmptyLines: true,
      complete: function (results) {
        const rowsArray = [];
        const valuesArray = [];

        // Iterating data to get column name and their values
          results.data.map((d) => {
          rowsArray.push(Object.keys(d));
          valuesArray.push(Object.values(d));
        });

        // Parsed Data Response in array format
        setParsedData(results.data);

        // Filtered Column Names
        setTableRows(rowsArray[0]);

        // Filtered Values
        setValues(valuesArray);
      },
    });
  };
  return (
    <div>
      {/* File Uploader */}
      <input
        type="file"
        name="file"
        onChange={changeHandler}
        accept=".csv"
        style={{ display: "block", margin: "10px auto" }}
      />
      <br />
      <br />
      <Table values = {values} tableRows = {tableRows}/>
      </div>
      );

  

}

export default App;