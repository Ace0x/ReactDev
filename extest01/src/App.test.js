import React from "react";
import { render, cleanup, waitFor } from "@testing-library/react";
import user from '@testing-library/user-event';

import App from "./App";

afterEach(cleanup);

it("should take a snapshot", () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});


const someValues = [{ name: 'teresa teng' }];

describe('68452480', () => {
  test('should pass', async () => {
    const { getByTestId, queryByTestId } = render(<App />);
    const str = JSON.stringify(someValues);
    const blob = new Blob([str]);
    const file = new File([blob], 'values.json', {
      type: 'application/JSON',
    });
    File.prototype.text = jest.fn().mockResolvedValueOnce(str);
    const input = getByTestId('upInput');
    user.upload(input, file);
    await waitFor(() => expect(queryByTestId('handler')).toBeTruthy());
  });
});
