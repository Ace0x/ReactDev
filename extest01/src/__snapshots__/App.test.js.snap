// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <input
      accept=".csv"
      name="file"
      style="display: block; margin: 10px auto;"
      type="file"
    />
    <br />
    <br />
    <table>
      <thead>
        <tr />
      </thead>
      <tbody />
    </table>
    <br />
    <table>
      <thead>
        <tr />
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
